### ISSUE TITLE
Get rid of sandboxes! (Drastic I know but bear with me)

### ISSUE THREAD

**Issue by smurphy8** [codeforokc/openbudgetokc#78]
I am proposing we get rid of the sandbox area for adding new features to the openbudget project.  
Sandboxes are very nice for individual developers to make contributions and I don't have a grudge against them in particular but I thought we should at least consider removing them.

# Proposal.  
Stop using sandboxes and instead develop features in place with branches for developers to work in.
#  How to implement
1. The sandbox directory would be renamed sandbox-deprecated.
2. We would point all new projects to branch the repo and develop in place.
# Reasons to take this action
## Less friction between development and release.
When features are being worked on in sandboxes and then released in other places, there will be a period in time where the routes have to be repointed at all the right places in order to be moved.  This means sources for assets like css and javascript will tend to be developed inside of the jade file instead of appropriate locations in the harp project.  That makes testing javascript routines harder.  

## Sanboxes become a source of versioning that is separate from git.
Because things in a sandbox are WIP there is a tendancy to plop useful files into the sandbox that then become confusing for developers looking on.  For instance the multiple versions of budget-tree currently in our sandbox.  It is hard for people looking on to know which file is the correct one.  By forcing this sort of wip stuff to be done in branches, the PR will be the point where cleanup is asked for.  This will keep our master branch cleaner. 

## Sandboxes make testing less likely
Features like cash-flow, radar and budget-tree could do with testing to make sure we don't add breaking changes. Doing tests in an enviornment where you know everything is going to be uprooted and moved is not very fun so they don't get done.

# Reasons to leave things as they are

## It works 
Sandboxes are an idea that makes intuitive sense.  Put some code somewhere and run it, work on it till you like it.

## Good for beginners
I think this is controversial but one way it is good for beginners is they don't have to branch git repos to do their work.  

## Clear lines between ready and not ready.  
Putting things in sandboxes mean you know that they aren't supposed to be part of the production release.  Even if something gets accidentally merged to master it won't make it to the release.

#  Lets drop it!

I don't think the reasons I can think of for keeping sandboxes outweigh problems.  I know I spent a lot of time working on the wrong version of the budget code because it was the one I saw working in the project.  I also had the wrong csv parsing because it was in the sandbox.  Beginners are going to struggle with branches regardless because we use a PR system.  If we do this though then it will be clearer where to go to make changes.

I think it will encourage slightly better development practices.

Anyway, that is all I have to say about it.  If everyone agrees I will do the work to fix it.  If not, cheers.

**Comment by thirdcreed** [comment]
Being new to the project but not to contributing, I can give my perspective:

I opened the project
I saw the instruction int he readme that said the sandbox is where we worked on visualizations
I saw that the sandbox had a sankey.js, and a flowable.html.
Flowable.html was broken. 
Made changes to sankey. 
Flowable still broken because it was trying to get data from csv, and csv was no longer present due to another change.
Saw note in readme that said that stuff might be broken.
Made a codepen instead, not being certain of the state of the sandbox.  

I don't have a strong opinion, i'm sure lots of good work was done on this idea.  

I will say that having a piece of your project that does not have the build requirement that it moves from one working state to another, can be confusing for someone who begins to develop later.  What is the correct state?  What's the correct data?  Am I misunderstanding something? Will I break the code I'm changing?  

If we do have a sandbox, it still needs IMO to have the requirement that it move from one stable build to another.
